lib/seahorn/BvOpSem2RawMemMgr.cc

```diff
@@ -308,21 +319,27 @@ Expr RawMemManager::coerce(Expr sort, Expr val) {
 /// \param[in] offsetBits number of bits at end of pointers reserved for byte
 ///            address
 /// \return symbolic value of the byte at the specified address
-Expr RawMemManager::extractUnalignedByte(Expr mem, PtrTy address,
-                                         unsigned offsetBits) {
+Expr RawMemManagerCore::extractUnalignedByte(MemValTy mem, const PtrTy &address,
+                                             unsigned offsetBits) {
   // pointers are partitioned into word address (high bits) and offset (low
   // bits)
-  PtrTy wordAddress = bv::extract(ptrSzInBits() - 1, offsetBits, address);
-  PtrTy byteOffset = bv::extract(offsetBits - 1, 0, address);
+  Expr wordAddress =
+      m_ctx.alu().Extract({address.toExpr(), ptrSizeInBits()},
+                          offsetBits /* low */, ptrSizeInBits() - 1 /* high */);
+  Expr byteOffset = m_ctx.alu().Extract({address.toExpr(), ptrSizeInBits()},
+                                        0 /* low */, offsetBits - 1 /* high */);
 
   // aligned ptr is address with offset bits truncated to 0
-  PtrTy alignedPtr =
-      bv::concat(wordAddress, bv::bvnum(0L, offsetBits, address->efac()));
+  PtrTy alignedPtr = PtrTy(
+      m_ctx.alu().Concat({wordAddress, ptrSizeInBits()} /* high */,
+                         {bv::bvnum(0L, offsetBits, address.toExpr()->efac()),
+                          ptrSizeInBits()} /* low */));
   Expr alignedWord = m_memRepr->loadAlignedWordFromMem(alignedPtr, mem);
 
-  byteOffset = bv::zext(byteOffset, wordSzInBits() - 3);
+  byteOffset = bv::zext(byteOffset, wordSizeInBits() - 3);
```

Comment:
for some reason, `zext`, `concat` and `extract` here are not used from `ALU`

Reply:
fixed